bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -187,6 +191,16 @@ func (b *BPFSniffer) ReadPacketData() ([]byte, gopacket.CaptureInfo, error) {
 	hdr := (*unix.BpfHdr)(unsafe.Pointer(&b.readBuffer[b.readBytesConsumed]))
 	frameStart := b.readBytesConsumed + int(hdr.Hdrlen)
 	b.readBytesConsumed += bpfWordAlign(int(hdr.Hdrlen) + int(hdr.Caplen))
+
+	if frameStart+int(hdr.Caplen) > len(b.readBuffer) {
+		captureInfo := gopacket.CaptureInfo{
+			Timestamp:     time.Unix(int64(hdr.Tstamp.Sec), int64(hdr.Tstamp.Usec)*1000),
+			CaptureLength: 0,
+			Length:        0,
+		}
+		return nil, captureInfo, nil
```

Comment:
Unclear. Why is this error happening? Perhaps a bug in the OpenBSD kernel. If we were to return an error then i'm not sure exactly what we'd say; perhaps that a BPF frame's header claims the frame is bigger than it is.